packages/cli/src/plugins/fetch.ts

```diff
@@ -0,0 +1,140 @@
+import { default as fse } from 'fs-extra'
+import type { RequestInfo, RequestInit, Response } from 'node-fetch'
+import { default as fetch_ } from 'node-fetch'
+import { join } from 'pathe'
+
+import { homedir } from 'os'
+import type { ContractConfig, ContractVersion, Plugin } from '../config'
+import type { RequiredBy } from '../types'
+
+export type FetchConfig<
+  TContract extends Omit<ContractConfig, 'path'> = Omit<ContractConfig, 'path'>,
+> = {
+  /**
+   * Duration in milliseconds to cache schemas from request.
+   *
+   * @default 1_800_000 // 30m in ms
+   */
+  cacheDuration?: number
+  /**
+   * Contracts to fetch schemas for.
+   */
+  contracts: TContract[]
+  /**
+   * Function for creating a cache key for contract.
+   */
+  getCacheKey(config: {
+    contract: TContract
+  }): `${string}:${string}:${string}:${ContractVersion}`
+  /**
+   * Name of source.
+   */
+  name?: TContract['name']
+  /**
+   * Function for parsing schema from fetch response.
+   *
+   * @default ({ response }) => response.json()
+   */
+  parse?({
+    response,
+  }: {
+    response: Response
+  }): Promise<unknown>
+  /**
+   * Function for returning a request to fetch schema from.
+   */
+  request(
+    config: Partial<TContract>,
+  ):
+    | Promise<{ url: RequestInfo; init?: RequestInit }>
+    | { url: RequestInfo; init?: RequestInit }
+
+  /**
+   * Duration in milliseconds before request times out.
+   *
+   * @default 5_000 // 5s in ms
+   */
+  timeoutDuration?: number
+}
+
+type FetchResult = RequiredBy<Plugin, 'contracts'>
+
+/**
+ * Fetches and parses contract schemas from network resource with `fetch`.
+ */
+export function fetch<
```

Comment:
I'd suggest to choose a name that doesn't clash with [native APIs](https://nodejs.org/docs/latest-v21.x/api/globals.html#fetch)